### PR TITLE
deepcopy config

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR fixes api_key being popped out never to return.

### CODE REVIEW

File: services/openai_analysis_service/app.py [hunk 1]
```diff
@@ -54,7 +55,7 @@
         log(f"index {summarize_embedding_index} created in pinecone")
     USE_PINECONE = True
 
-summary_config = cfg['analysis_services.openai_analysis_service.summary']
+summary_config = copy.deepcopy(cfg['analysis_services.openai_analysis_service.summary'])
```
Comment (guynir42) [inline services/openai_analysis_service/app.py:58]: I think you can just do `cfg.copy([...])` and you get the same thing. But I'm not against an explicit "deep copy" command either.

File: skyportal/handlers/api/analysis.py [hunk 2]
```diff
@@ -54,6 +57,18 @@
 
 DEFAULT_ANALYSES_DAILY_LIMIT = 100
 
+# check for API key
+summary_config = copy.deepcopy(cfg['analysis_services.openai_analysis_service.summary'])
+if summary_config.get("api_key"):
+    # there may be a global API key set in the config file
+    openai_api_key = summary_config.get("api_key")
+    summary_config.pop("api_key")
```
Comment (guynir42) [inline skyportal/handlers/api/analysis.py:65]: Can we combine these two lines into just "pop"?

File: static/js/components/SourceDesktop.jsx [hunk 3]
```diff
@@ -298,6 +298,8 @@ const SourceDesktop = ({ source }) => {
     </Tooltip>
   );
 
+  console.log("source", source);
```
Comment (guynir42) [inline static/js/components/SourceDesktop.jsx:301]:
```suggestion
```

**Review by guynir42 — APPROVED** [review]
I have minor comments.

File: skyportal/handlers/api/analysis.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/analysis.py:65]: We use the summary_config later on.

File: services/openai_analysis_service/app.py (hunk 1 above)
Comment (mcoughlin) [reply services/openai_analysis_service/app.py:58]: I think it needs to be a deepcopy (copy is shallow).

File: skyportal/handlers/api/analysis.py (hunk 2 above)
Comment (guynir42) [reply skyportal/handlers/api/analysis.py:65]: yeah, do `openai_api_key=summary_config.pop('api_key')`

File: services/openai_analysis_service/app.py (hunk 1 above)
Comment (guynir42) [reply services/openai_analysis_service/app.py:58]: I think dictionary `copy` is deep by default, but I'm going off my memory.

Comment (guynir42) [reply services/openai_analysis_service/app.py:58]: Yeah, you are right.

File: skyportal/handlers/api/analysis.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/analysis.py]: Ah got ya. Done.

File: services/openai_analysis_service/app.py (hunk 1 above)
Comment (guynir42) [reply services/openai_analysis_service/app.py:58]: Although if you are only popping the top level keys I'm not sure it matters. 
In any case, `deepcopy` is more explicit so I think it is fine.

File: static/js/components/SourceDesktop.jsx (hunk 3 above)
Comment (mcoughlin) [reply static/js/components/SourceDesktop.jsx]: Done.

**Pull request merged**